tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/LanguagePluginAbstract.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+package fi.helsinki.cs.tmc.langs;
+
+import java.nio.file.Path;
+
+/**
+ *
+ * @author TMC-langs
+ */
+public abstract class LanguagePluginAbstract implements LanguagePlugin {
+
+    protected final String languageName;
```

Comment:
Seems unnecessary. I see no benefit over having the subclass just override getLanguageName() itself.